forum/src/main/java/unipd/se18/ocrcamera/forum/ForumLogin.java

```diff
@@ -0,0 +1,182 @@
+
+        //Definition of view model listener
+        viewModel.setForumLoginListener(new Login_VM.ForumLoginListener() {
+
+            /**
+             * The method of the listener is triggered when the database response
+             * to the login request is positive and the view model requests to load
+             * the ShowPost fragment, so that the user can access the forum contents
+             *
+             * @param username The username of the user that successfully logged in
+             * @author Alberto Valente (g2), Taulant Bullaku (g2)
+             */
+            @Override
+            public void onLoginSuccess(String username) {
+
+                if(username != null) {
+                    //creates the username bundle
+                    Bundle bundle = new Bundle();
+                    bundle.putString(KEY_USERNAME, username);
+
+                    //creates an instance of the fragment to be launched
+                    ShowPosts showPostsFragment = new ShowPosts();
+                    //passes the bundle to the fragment as an argument
+                    showPostsFragment.setArguments(bundle);
+
+                    //performs the fragment transaction
+                    getActivity()
+                            .getSupportFragmentManager()
+                            .beginTransaction()
+                            .replace(
+                                    R.id.fragmentContainer,
+                                    showPostsFragment
+                            )
+                            .commit();
+                }
+                else {
+                    Log.d(LOG_TAG, LOG_NULL_USERNAME);
+                }
+            }
+
+            /**
+             * The method of the listener is triggered when the database response to
+             * the login request is negative and the view model requests to show the
+             * user an error message, that is performed through the use of a toast
+             *
+             * @param message The error message about what was wrong with the login request
+             * @author Alberto Valente (g2), Taulant Bullaku (g2)
+             */
+            @Override
+            public void onLoginFailure(String message) {
+
+                if (message != null) {
+                    Toast.makeText(getContext(), message, Toast.LENGTH_LONG).show();
+                }
+                else {
+                    Log.d(LOG_TAG, LOG_NULL_ERROR_MESSAGE);
+                }
+            }
+        });
+
+        //When the login button is clicked a request is sent through the view model
+        loginButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                //The username and password inserted are gathered from the EditText objects
+                userName = usernameEditText.getText().toString();
+                userPwd = pwdEditText.getText().toString();
```

Comment:
I think that could be better to not let the password as a normal string in the code for security reason. i suggest u to cript it before use and send it to Login_VM. i know this is a base implementation so this is an advice for future improvement.

Reply:
You're totally right: I'm going to implement this as soon as possible because, as you said, is a fairly basic security issue.